server/aesdsocket.c

```diff
@@ -67,227 +110,348 @@ int deamon_create()
+		printf("Malloc Error \n");
+      		exit(-1);
+	
+	}
+	memset(receive_buffer,0,BUFFER_SIZE);
+ 
+        memset(temp_buffer, 0, BUFFER_SIZE);
+        
+        while (!packet_complete)
+        	{
+        		receive_bytes = recv(thread_data->accept_socket_fd, temp_buffer, BUFFER_SIZE, 0);
+        		if (receive_bytes == -1)
+        		{
+        			syslog(LOG_ERR,"Accept command error");
+        			printf("Receive comamnd error\n");
+        			exit(-1);
+        		}
+			else{
+			final_size++;
+			for (byte_count = 0; byte_count < BUFFER_SIZE; byte_count++)
+			{
+				if(temp_buffer[byte_count] == '\n')			
+				{
+					packet_complete = true;
+					syslog(LOG_DEBUG,"String received is %s",temp_buffer);
+					break;
+				}
+			
+			}
+			}
+		        
+		        total_packet_size = total_packet_size + receive_bytes;
+		        receive_buffer = (char *) realloc (receive_buffer, total_packet_size + 1);
+		        if(final_size >=BUFFER_SIZE) 
                 {
                   if(!(chec_ptr = realloc(receive_buffer, BUFFER_SIZE + final_size + 1)))
                     {
                       syslog(LOG_ERR , "realloc\n");
-                      return -1;
+                      exit(-1);
                     }
                   else
                     {
                       receive_buffer = chec_ptr;
-                      strncat(receive_buffer, recv_buffer, final_size);
+                      strncat(receive_buffer, temp_buffer, final_size);
                     }
 
                 }
-              if(!packet_check_flag)    //Check for packet check flag status
-                {
-                  receive_buffer = (char *)realloc(receive_buffer , (sizeof(char) * (BUFFER_SIZE * final_size)));
+		        if (receive_buffer == NULL)
+		        {
+		        	syslog(LOG_ERR,"Realloc() Error");
+        			printf("Realloc() Error \n");
+        			break;
+		        } 
+		        strncat(receive_buffer, temp_buffer, receive_bytes);
+			syslog(LOG_DEBUG,"Appended String is %s",receive_buffer);
+			}
+			syslog(LOG_DEBUG,"Appended String is %s",receive_buffer);
+		        open_fd = open("/var/tmp/aesdsocketdata", O_RDWR | O_APPEND , 0644);
+			if (open_fd == -1)
+			{
+				syslog(LOG_ERR,"File cannot be opened in WRITE mode");
+				perror("WRITE");
+				exit(-1);
+			}
+			syslog(LOG_DEBUG, "OPEN status %d", open_fd);
+			status = pthread_mutex_lock(thread_data->mutex);
+			if(status)
+			{
+				syslog(LOG_ERR, "ERROR: mutex_lock() fail");
+		
+			}
+	
+			lseek(open_fd, 0, SEEK_END);
+			status = write(open_fd, receive_buffer, strlen (receive_buffer));
+			if (status == -1)
+			{
+				syslog(LOG_ERR,"The received data is not written ");
+				exit(-1);
+			}
+			status = pthread_mutex_unlock(thread_data->mutex);
+			if(status) 
+			{
+			syslog(LOG_ERR, "ERROR: mutex_lock() fail");
+			}
+			int file_size = lseek(open_fd, 0, SEEK_CUR);
+			lseek(open_fd, 0, SEEK_SET);
+			send_buffer = (char *) malloc(file_size * sizeof(char));
```

Comment:
@Rajs060699 what assumptions are you making here about the size of the heap vs the size of all packets sent?